feat: onboarding screen

FilipiRafael opened the pull request:
### ✨ Description:
Develop and design the new onboarding screen based on the new app branding

### 🛠 Changes

- [X] : add a `.prettierignore` file
- [X] : add `isOnboardingComplete` MMKV store value to handle app routes
- [X] : refactor onboarding screen to small components and add new items to `constants.ts`
- [X] : replace onboarding images and use `FastImage` to improve images's performance
- [X] : update onboarding title copy on the i18n dictionary
- [X] : install `react-native-skia` and use it for creating the `OnboardingMask` animation

### 🧪 How Has This Been Tested?

- [X] Tested on iOS
- [x] Tested on Android

### 📸 How it's looking:
https://github.com/user-attachments/assets/edf7beb3-ffd0-4d18-99ab-dfb753702ecb

FilipiRafael committed:
chore: remove onboarding description from i18n

FilipiRafael committed:
chore: replace onboarding screen images and add logo image

FilipiRafael committed:
chore: remove deleted images from favoritesHeader and Home screen

FilipiRafael committed:
chore: add prettier ignore

FilipiRafael committed:
chore: add vscode workspace settings to prettier ignore and run prettier on its file

FilipiRafael committed:
feat: update onboarding screen and install skia and skottie with patch-package

FilipiRafael committed:
chore: update onboarding screen

FilipiRafael committed:
chore: remove unsed pictures

FilipiRafael committed:
feat: add onboarding titles to i18n and onboardign animations

FilipiRafael committed:
chore: add .json to prettierignore

FilipiRafael committed:
feat: add react-native-haptic-feedback

FilipiRafael committed:
feat: add meow sound when finishing the onboarding

FilipiRafael committed:
chore: add haptic feedback to constants file

FilipiRafael committed:
feat: replace onboarding assets

FilipiRafael committed:
chore: remove audio files and handlePlaySound on the onboardingScreen

FilipiRafael committed:
fix: add button margin top between title

FilipiRafael committed:
feat: use touchableOpacity instead of pressable on button

FilipiRafael committed:
feat: onboarding animation and remove safeAreaView from App.tsx

FilipiRafael committed:
fix: onboarding screen responsive design

FilipiRafael committed:
chore: update onboarding images and add a slower animation

FilipiRafael committed:
refactor: move contants from onboarding screen to contants.ts

FilipiRafael committed:
refactor: move mask to onboardingMask component

FilipiRafael committed:
fix: set onboardingMask in the first tre render

FilipiRafael committed:
refactor: move pagination to onboardingPagination

FilipiRafael committed:
refactor: move button to OnboardingButton component

FilipiRafael committed:
chore: changes after running prettier:write

FilipiRafael committed:
refactor: reorganize imports by origin

FilipiRafael committed:
fix: update onboarding copy in i18n dictionary

FilipiRafael committed:
chore: update pt and es onboarding copy on i18n dictionary

FilipiRafael committed:
feat: update onboarding wrapper to better show the title

FilipiRafael committed:
feat: create OnboardingImages component and use FastImage instead of Image from react-native

FilipiRafael committed:
chore: remove react-native-skottie and its patch-package

FilipiRafael committed:
fix: remove local images fast image preloading

FilipiRafael committed:
feat: make the onboarding image transition smooth

gustavoabel commented on app/config/constants.ts:
@FilipiRafael Why are we exporting constants from the onboarding component in app/config/constants? It is not necessary, we will only do this for constants that are used in more than one place. Also, constants that are component-specific should be created inside their respective components.

FilipiRafael replied:
Good to know @gustavoabel, so I already moved the component-specific constants to the `OnboardingScreen.tsx`, now I'm exporting them from there.

FilipiRafael committed:
feat: update onboarding image background

FilipiRafael committed:
fix: add loading validation to avoid fast clicks broking the animation

FilipiRafael committed:
refactor: reorder states

FilipiRafael committed:
refactor: reorder imports based on size

FilipiRafael committed:
refactor: move component-specific constants to OnboardingScreen and exporting them from there

FilipiRafael committed:
style: changes after running .eslintrc.js 46ms
.github/pull_request_template.md 26ms
.prettierrc.js 3ms
App.tsx 123ms
app/components/BottomSheet/BottomSheet.tsx 13ms
app/components/BottomSheet/styles.ts 3ms
app/components/FavoritesHeader/FavoritesHeader.tsx 2ms
app/components/FavoritesHeader/styles.ts 2ms
app/components/index.ts 1ms
app/components/OnboardingBackground/OnboardingBackground.tsx 3ms
app/components/OnboardingBackground/styles.ts 2ms
app/components/OnboardingButton/OnboardingButton.tsx 11ms
app/components/OnboardingButton/styles.ts 2ms
app/components/OnboardingImages/OnboardingImages.tsx 5ms
app/components/OnboardingImages/styles.ts 2ms
app/components/OnboardingMask/OnboardingMask.tsx 5ms
app/components/OnboardingMask/styles.ts 1ms
app/components/OnboardingPagination/OnboardingPagination.tsx 4ms
app/components/OnboardingPagination/styles.ts 2ms
app/config/constants.ts 4ms
app/config/i18n.ts 2ms
app/config/platform.ts 2ms
app/config/storage.ts 1ms
app/resources/index.ts 1ms
app/resources/strings/en.ts 11ms
app/resources/strings/es.ts 9ms
app/resources/strings/pt.ts 11ms
app/routes/index.ts 2ms
app/routes/Routes.tsx 3ms
app/screens/FavoritesScreen/FavoritesScreen.tsx 10ms
app/screens/FavoritesScreen/styles.ts 3ms
app/screens/HomeScreen/HomeScreen.tsx 8ms
app/screens/HomeScreen/styles.ts 3ms
app/screens/index.ts 1ms
app/screens/OnboardingScreen/OnboardingScreen.tsx 6ms
app/screens/OnboardingScreen/styles.ts 2ms
babel.config.js 2ms
index.js 3ms
jest.config.js 2ms
metro.config.js 2ms
react-native.config.js 1ms
README.md 10ms

FilipiRafael committed:
feat: update onboarding background images

FilipiRafael committed:
chore: change haptic feedback trigger location

FilipiRafael committed:
chore: update background and cat illustrations

gustavoabel reviewed (approved):
LGTM
